File: algorithms/1013. Partition Array Into Three Parts With Equal Sum/solution.py
class Solution:
    def canThreePartsEqualSum(self, A: List[int]) -> bool:
        total, cur, counter = sum(A), 0, 0
        if total % 3 != 0:
            return False
        for n in A:
            cur += n
            if cur == total / 3:
                counter += 1
                if counter == 3:
                    return True
                cur = 0
        return False

File: algorithms/1013. Partition Array Into Three Parts With Equal Sum/solution.cc
class Solution {
public:
    bool canThreePartsEqualSum(vector<int>& A) {
        int sum = accumulate(A.begin(), A.end(), 0), counter = 0;
        if (sum % 3 != 0)
            return false;
        for (int i = 0, cur = 0; i < A.size() && counter < 3; ++i) {
            cur += A[i];
            if (cur == sum / 3) {
                ++counter;
                cur = 0;
            }
        }
        return counter >= 3;
    }
};
